src/components/topicform/topicform.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from "react";
+import { Button, Form } from "react-bootstrap";
+import TagsInput from "react-tagsinput";
+import "./topicform.scss";
+import "react-tagsinput/react-tagsinput.css";
+
+class TopicForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      TopicName: "",
+      TopicDescription: "",
+      tags: [],
+    };
+  }
+
+  handleTopicSubmission = () => {};
+
+  handleChangeTags = (tags) => {
+    this.setState({ tags });
+    console.log(tags);
+  };
+
+  handleChange = (params, event) => {
+    event.preventDefault();
+    if (params === "TopicName") {
+      this.setState({ TopicName: event.target.value });
+    } else if (params === "TopicDescription") {
+      this.setState({ TopicDesctiption: event.target.value });
```

Comment:
I think we don't need two way binding in this component as we are not starting with blanks and not some initial state, so upon submission the value of input fields in the form can be retrieved via the event object or some other means. So maybe we can turn this into a functional component as well.

Reply:
But controlled components are the best way to handle the forms?